File: p002/main.jl
#!/usr/bin/env julia

function main()
  sum = 0
  prev, curr = 1, 2
  while curr < 4000000
    if curr % 2 == 0
      sum += curr
    end

    prev, curr = curr, prev + curr
  end

  println(sum)
end

main()

File: p002/main.rs
fn main() {
    let mut sum = 0;
    let mut prev = 1;
    let mut curr = 2;

    while curr <= 4_000_000 {
        if curr % 2 == 0 {
            sum += curr
        }

        (prev, curr) = (curr, prev + curr);
    }
    println!("{}", sum);
}
